### PR TITLE
[`PEFT`] Fix scale unscale with LoRA adapters

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Fixes: https://github.com/huggingface/diffusers/issues/5414

Needs: https://github.com/huggingface/peft/pull/1029

Before this PR, weighted multi adapter inference was broken because `module.scale_layer` was initializing the scale of the module with its original value instead of multiplying the existing value.

The fix is to

1- Multiply the existing scale with the new scale factor 
2- unscale by dividing the scale with the scale factor to retrieve the correct value.
3- use `set_scale` to get the correct scale ( alpha / r ) --> this method is used inside `set_weights_and_activate_adapters`

cc @pacman100 @BenjaminBossan @patrickvonplaten

### CODE REVIEW

**Review by sayakpaul — APPROVED** [review]
Thanks much! 

I think we need to rerun the code of the [docs](https://huggingface.co/docs/diffusers/main/en/tutorials/using_peft_for_inference) to make sure the results are correct, right? 

Good to merge for me after the conflicts have been resolved.

File: src/diffusers/utils/peft_utils.py
Comment (pacman100) [inline src/diffusers/utils/peft_utils.py:95]:
```suggestion
def unscale_lora_layers(model, weight: Optional[float] = None):
```

File: src/diffusers/utils/peft_utils.py
Comment (pacman100) [inline src/diffusers/utils/peft_utils.py:102]: This weight needs to be passed to `module.unscale_layer()` in line 110

File: src/diffusers/utils/peft_utils.py
Comment (pacman100) [inline src/diffusers/utils/peft_utils.py:189]: What happens if `adapter_name` is not available for this lora layer? Is this handled in PEFT to do nothing if adapter is unavailable?

**Review by pacman100 — COMMENTED** [review]
Thank you @younesbelkada for working on this 🤗, left a few comments.

File: src/diffusers/utils/peft_utils.py
Comment (sayakpaul) [reply src/diffusers/utils/peft_utils.py:189]: I think it errors it out from `peft` in case that is hit. @younesbelkada can confirm. 

From what I recollect, the users are required to provide `adapter_name` when dealing with multiple LoRAs otherwise, for the single case, the default naming is used.

Comment (younesbelkada) [reply src/diffusers/utils/peft_utils.py:195]: https://github.com/huggingface/peft/pull/1034 should take care of that

**Review by pacman100 — APPROVED** [review]
Thank you @younesbelkada for the fixes, LGTM! 🚀
As Sayak mentioned, could you please rerun the docs and notebook example and see that things are working as expected.

File: src/diffusers/utils/peft_utils.py
Comment (sayakpaul) [reply src/diffusers/utils/peft_utils.py:195]: https://github.com/huggingface/peft/pull/1034 is merged. So, no blockers.

File: tests/lora/test_lora_layers_peft.py
Comment (sayakpaul) [inline tests/lora/test_lora_layers_peft.py:1181]: @younesbelkada why are these values being changed?

File: tests/lora/test_lora_layers_peft.py
Comment (sayakpaul) [inline tests/lora/test_lora_layers_peft.py:1195]: @younesbelkada why are these values being changed?

**Review by sayakpaul — APPROVED** [review]
LGTM! Once the PR is merged, I will take care of the docs.

File: tests/lora/test_lora_layers_peft.py
Comment (younesbelkada) [reply tests/lora/test_lora_layers_peft.py:1181]: because the previous API was broken and the values were wrong :/ so this PR updates the slow tests with the correct values

**Comment by younesbelkada** [comment]
Thanks @sayakpaul !

**Comment by sayakpaul** [comment]
Regarding the failing "inconsistency test", indeed, it seems to be weird to me since running `make fix-copies` doesn't yield any changes for me locally.

**Comment by younesbelkada** [comment]
the test is also failing on main: https://github.com/huggingface/diffusers/actions/runs/6566415588/job/17837040326 perhaps we can merge this PR as is then

**Comment by sayakpaul** [comment]
@patrickvonplaten could you give this a review too?

**Review by patrickvonplaten — APPROVED** [review]
Thanks!

**Pull request merged**
